### PR TITLE
[server / inferrer] prefetch configuration files

### LINKED ISSUE #5527: Improve Inferrer Performance
See https://github.com/gitpod-io/gitpod/pull/5474#discussion_r701848907

### PR DESCRIPTION
**Pull request by svenefftinge**
Prefetch configuration files in parallel and in advanced based on a static cache.

Context: The inferred logic is based on a conditional tree, that asks about configuration files sequentially one by one. Requesting all those files in parallel should speed up the process.

fixes #5527

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **svenefftinge** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **svenefftinge** after the PR has been reviewed.

Associated issue: *#5527*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/projects/projects-service.ts
Comment (gtsiolis) [inline components/server/src/projects/projects-service.ts:204]: **nitpick**: Minor typo in the comment.

```suggestion
        // eagerly fetch for all files that the inferrer usually asks for.
```

**Review by gtsiolis — APPROVED** [review]
Didn't notice any difference in terms of UX. LGTM! ✔️ 

Approving to unblock merging this but holding a) because of one minor typo in the comments and b) in case @AlexTugarev wants to take a closer look at the code. 🏓 

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: d68c7a7ce5a760e435aede1e5136b5eb7255ba8e</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5529#pullrequestreview-747197847" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **svenefftinge** after the PR has been reviewed.

Associated issue: *#5527*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by AlexTugarev** [comment]
/approve 

That make much sense!

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5529#issuecomment-913668702" title="Approved">AlexTugarev</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5529#pullrequestreview-747197847" title="Approved">gtsiolis</a>*

Associated issue: *#5527*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [AlexTugarev]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/projects/projects-service.ts
Comment (AlexTugarev) [inline components/server/src/projects/projects-service.ts:197]: 👍🏻

**Pull request merged**
